src/Search/Processors/SearchUpdateProcessor.php

```diff
@@ -83,12 +83,19 @@ protected function prepareIndexes()
                 SearchVariant::activate_state($state);
 
                 // Ensure that indexes for all new / updated objects are included
+                $indexesWithDeletes = [];
                 $objs = DataObject::get($base)->byIDs(array_keys($ids));
```

Comment:
Unfortunately $base is going to be SiteTree, this means that any subclasses (e.g. MyPage) will be retrieved as SiteTree instead, so MyPage->getShowInSearch() won't be called :-(

This happens even if I explicitly add this to the index:
$this->addClass(MyPage::class);

Reply:
`addDirtyIDs ()` has this: `$base = DataObject::getSchema()->baseDataClass($class);`, which is then used in `prepareIndexes()`. So at some point, the specific class was available in this implementation, it's just a matter of surfacing it in the intermediary array?

Reply:
yup ignore me, I'm not sure how I came to this conclusion.  I've written a unit test that explicitly tests for getShowInSearch() on a couple of classes and that's passing